apps/studio/src/features/editing-experience/components/form-builder/renderers/controls/JsonFormsObjectControl.tsx

```diff
@@ -0,0 +1,61 @@
+import {
+  Generate,
+  findUISchema,
+  isObjectControl,
+  rankWith,
+  type RankedTester,
+  type StatePropsOfControlWithDetail,
+} from '@jsonforms/core'
+import { JsonFormsDispatch, withJsonFormsDetailProps } from '@jsonforms/react'
+import { useMemo } from 'react'
+import { JSON_FORMS_RANKING } from '~/constants/formBuilder'
+
+export const jsonFormsObjectControlTester: RankedTester = rankWith(
+  JSON_FORMS_RANKING.ObjectControl,
+  isObjectControl,
+)
+
+export function JsonFormsObjectControl({
+  path,
+  visible,
+  renderers,
+  cells,
+  schema,
+  enabled,
+  uischema,
+  uischemas,
```

Comment:
why got 2 schemas

Reply:
The first one is the specific UI schema defined for the particular element that it is rendering. The second one refers to the registry of all UI schema elements (similar to the control renderers registry we define in FormBuilder.tsx).

We don't actually use this, the code is mostly adapted from their material renderer and the `findUISchema` method signature requires it. Basically what this does is that it tries to generate a sub-UISchema within the context of the object that it is operating on, which is then dispatched and handled by JSONForms to render the elements within this object.

Reply:
> specific UI schema defined for the particular element that it is rendering

we don't define ui schemas for elements right?

Reply:
Nope, but JSONForms generates one automatically, which is then used to control the rendering of the form fields and the layouts.